vernac/comInductive.ml

```diff
@@ -375,7 +375,9 @@ let interp_mutual_inductive_constr ~sigma ~template ~udecl ~ctx_params ~indnames
       indnames arities arityconcl constructors
   in
   let template = List.map4 (fun indname (templatearity, _) concl (_, ctypes) ->
-      let template_candidate () =
+      let template_candidate () = match concl with
+      | Some (Prop | Set | SProp) -> false
+      | None | Some (Type _) ->
```

Comment:
What about records? Shouldn't this test be in the `template_polymorphism_candidate` function?

Reply:
I have further commits cleaning this piece of code, because there are a lot of hidden invariants going on. I have to fix this PR first though.